src/App.tsx

```diff
@@ -1,39 +1,62 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  clock: string;
+  hasClock: boolean;
+};
 
-        <span className="Clock__time">
-          {today.toLocaleTimeString()}
-        </span>
+type Props = {};
+
+export class App extends Component<Props, State> {
+  state = {
+    clock: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerName = 0;
+
+  componentDidMount() {
+    document.addEventListener('click', this.leftClick);
+    document.addEventListener('contextmenu', this.rightClick);
+    this.timerName = window.setInterval(() => {
+      this.setState({
+        clock: getRandomName(),
+      });
+    }, 3300);
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerName);
+    document.removeEventListener('click', this.leftClick);
+    document.removeEventListener('click', this.rightClick);
+  }
+
+  leftClick = () => {
+    this.setState({ hasClock: true });
+  };
+
+  rightClick = (event: MouseEvent) => {
+    this.setState({ hasClock: false });
+    event.preventDefault();
+  };
```

Comment:
Still actual [checklist point 3](https://github.com/mate-academy/react_clock/blob/master/checklist.md)
```suggestion
  handleLeftClick = () => {
    this.setState({ hasClock: true });
  };

  handleRightClick = (event: MouseEvent) => {
    this.setState({ hasClock: false });
    event.preventDefault();
  };
```